src/databricks/labs/ucx/azure/resources.py

```diff
@@ -242,3 +242,36 @@ def _role_name(self, role_definition_id) -> str | None:
                 return None
             self._role_definitions[role_definition_id] = role_name
         return self._role_definitions.get(role_definition_id)
+
+    def managed_identity_client_id(
+        self, access_connector_id: str, user_assigned_identity_id: str | None = None
+    ) -> str | None:
+        # get te client_id/application_id of the managed identity used in the access connector
+        try:
+            identity = self._get_resource(access_connector_id, "2023-05-01").get("identity")
+        except NotFound:
+            logger.warning(f"Access connector {access_connector_id} no longer exists")
+            return None
+
+        if not identity:
+            return None
+        if identity.get("type") == "UserAssigned":
+            if not user_assigned_identity_id:
+                return None
+            identities = identity.get("userAssignedIdentities")
+            if user_assigned_identity_id in identities:
+                return identities.get(user_assigned_identity_id).get("clientId")
+            # sometimes we see "resourceGroups" instead of "resourcegroups" in the response from the API
+            if user_assigned_identity_id.replace("resourcegroups", "resourceGroups") in identities:
+                return identities.get(user_assigned_identity_id.replace("resourcegroups", "resourceGroups")).get(
```

Comment:
Nit: extract a var to make more readable

Reply:
fixed.